bin/cli.js

```diff
@@ -101,29 +102,35 @@
                     method: req.method,
                     headers: {
                         "Content-Length": req.headers["content-length"] || 0,
-                        "Content-Type": req.headers["content-type"],
-                        "Authorization": req.headers["authorization"]
+                        "Content-Type": req.headers["content-type"] || '',
+                        "Authorization": req.headers["authorization"] || ''
                     },
                     followAllRedirects: true,
-                    body: body,
+                    body: body || '',
                     jar: false,
-                    strictSSL: false
+                    strictSSL: false,
+                    rejectUnauthorized : false
                 };
 
                 try {
-                    request(options, function(err, response, data) {
+                    needle.request(options.method, options.url, options.body, options, function(err, response, body)
+                    {
                         try {
                             var statusCode = (response ? response.statusCode : 500) || 500;
                             var headers = (response ? response.headers : {}) || {};
 
+                            // Used to make the body of Needle as same as Request library.
+                            var bodyString = JSON.stringify(body).replace(/<[/]/g, "<\\/");
```

Comment:
I think maybe the preferred way to address this would be to tell needle not to parse this as JSON to begin with, it looks like the default behavior is to have this on - can we turn parsing responses off and hopefully save some performance while getting the same behavior as the `request` library? https://www.npmjs.com/package/needle#response-options

Reply:
Suggested changes have been made and tests are passing with new changes.

Reply:
Great, @zenmoto can you double check our approach here? I want to make sure we get this right since it's a change to the core client.

Reply:
Sorry ignore that last comment this is a change to the CLI.